test/sync_tests/conflicting.test.js

```diff
@@ -0,0 +1,763 @@
+    await this.application.syncService.sync(syncOptions);
+    expect(this.application.itemManager.items.length).to.equal(
+      this.expectedItemCount
+    );
+
+    // client B
+    await this.application.syncService.clearSyncPositionTokens();
+
+    // This client says this item is deleted, but the server is saying its not deleted.
+    // In this case, we want to keep the server copy.
+    await this.application.changeAndSaveItem(
+      note.uuid,
+      (mutator) => {
+        mutator.setDeleted();
+        mutator.updated_at = Factory.yesterday();
+      },
+      undefined,
+      undefined,
+      syncOptions
+    );
+
+    // We expect that this item maintained.
+    expect(this.application.itemManager.items.length).to.equal(
+      this.expectedItemCount
+    );
+  });
+
+  it('should create conflict if syncing an item that is stale', async function () {
+    let note = await Factory.createMappedNote(this.application);
+    await this.application.itemManager.setItemDirty(note.uuid);
+    await this.application.syncService.sync(syncOptions);
+    note = this.application.findItem(note.uuid);
+    expect(note.dirty).to.equal(false);
+    this.expectedItemCount++;
+    /** First modify the item without saving so that
+     * our local contents digress from the server's */
+    await this.application.changeItem(note.uuid, (mutator) => {
+      mutator.title = `${Math.random()}`;
+    });
+    note = await this.application.changeAndSaveItem(
+      note.uuid,
+      (mutator) => {
+        mutator.text = 'Stale text';
+        mutator.updated_at = Factory.yesterday();
+      },
+      undefined,
+      undefined,
+      syncOptions
+    );
+    expect(note.dirty).to.equal(false);
+
+    // We expect now that the item was conflicted
+    this.expectedItemCount++;
+
+    const rawPayloads = await this.application.storageService.getAllRawPayloads();
+    expect(rawPayloads.length).to.equal(this.expectedItemCount);
+    for (const payload of rawPayloads) {
+      expect(payload.dirty).to.not.be.ok;
+    }
+  });
+
+  it('creating conflict with exactly equal content should keep us in sync', async function () {
+    const note = await Factory.createMappedNote(this.application);
+    await this.application.itemManager.setItemDirty(note.uuid);
+    this.expectedItemCount++;
+
+    await this.application.syncService.sync(syncOptions);
+
+    await this.application.changeAndSaveItem(
+      note.uuid,
+      (mutator) => {
+        mutator.updated_at = Factory.yesterday();
+      },
+      undefined,
+      undefined,
+      syncOptions
+    );
+
+    expect(this.application.itemManager.items.length).to.equal(
+      this.expectedItemCount
+    );
+  });
+
+  it('handles stale data in bulk', async function () {
+    /** This number must be greater than the pagination limit per sync request.
+     * For example if the limit per request is 150 items sent/received, this number should
+     * be something like 160. */
+    const largeItemCount = 160;
```

Comment:
Any way to access this constant from the application code? As in, have something like:
```ts
const largeItemCount = this.application.syncService.largeitemCount + 1
```
This would prevent some test/code drifting